go/iot/protocol/examples/arduino/protocol.ino

```diff
@@ -154,11 +172,65 @@ JsonObject* parseResponse(char *json) {
     //Serial.println("parseObject() failed");
     return NULL;
   } else{
-    return &root;
+    response *res = new response(root["Code"], root["Info"]);
+    return res;
   }
 }
+// Typical example is {"Code" : 200, "Info" : "TMP"}
+// There could be whitespace inside the message
+response* parseResponseText(char *text) {
+  String ss(text);
+  String codeKey = "\"Code\"";
+  String infoKey = "\"Info\"";
+  int codeKeyPos = ss.indexOf(codeKey);
+  int infoKeyPos = ss.lastIndexOf(infoKey);
+  if (codeKeyPos == -1 || infoKeyPos == -1)
+    return NULL;
+  codeKeyPos = codeKeyPos + codeKey.length();
+  infoKeyPos = infoKeyPos + infoKey.length();
+  // Find the code value
+  // Value is between the ',' and ':'
+  int colonIndex = codeKeyPos;
+  while(colonIndex < ss.length() && ss[colonIndex] != ':') colonIndex++;
+  if (colonIndex == ss.length()) return NULL;
+  int commaIndex = colonIndex;
+  while(commaIndex < ss.length() && ss[commaIndex] != ',') commaIndex++;
+  if (commaIndex == ss.length()) return NULL;
+  String valuess = ss.substring(colonIndex + 1, commaIndex);
+  valuess.trim();
+  int value = toInteger(valuess);
+  if (value == -1) return NULL;
+
+  //exclusive the right bracket
+  String info = ss.substring(infoKeyPos + 1, ss.length() - 1);
+  int leftMark = info.indexOf('\"');
+  int rightMark = info.lastIndexOf('\"');
+  if (leftMark == -1 || rightMark == -1 || leftMark == rightMark)
+    return NULL;
+  info = info.substring(leftMark + 1, rightMark);
+
+  response *res = new response(value, info);
+  return res;
+}
+
+int toInteger(String code){
```

Comment:
Isn't toInt working for your case? https://www.arduino.cc/en/Tutorial.StringToIntExample

Reply:
toInt would return 0 for those invalid value.  And for string like "123s123" it would return 123.